### PR TITLE
#493 Add CRAPL license

### PR DESCRIPTION
**Pull request by yasharmaster**
Added CRAPL license for issue #493 
@pombredanne please review

### CODE REVIEW

**Comment by pombredanne** [comment]
@yasharmaster Thanks! do you mind updating your commit message as per https://github.com/nexB/aboutcode/wiki/Writing-good-commit-messages?

File: src/licensedcode/data/licenses/crapl-0.1.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+key: crapl-0.1
+short_name: crapl-0.1
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/crapl-0.1.yml]: `CRAPL v0 BETA 1` would be a better name

File: src/licensedcode/data/licenses/crapl-0.1.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+name: CRAPL v0 BETA 1
+category: CRAPL
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/crapl-0.1.yml]: This is not a category. Please look at other licenses categories. We may need @DennisClark take on this

**Comment by yasharmaster** [comment]
@pombredanne Should I send a new PR with updated commit messages and the requested changes?

**Comment by pombredanne** [comment]
thanks for the fixes. I will review once CI tests are green

File: src/licensedcode/data/licenses/crapl-0.1.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+key: crapl-0.1
+short_name: CRAPL v0 BETA 1
+name: CRAPL v0 BETA 1
+category: Academic
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/crapl-0.1.yml]: Thank for this and the other changes. I need @DennisClark input on the category before I can merge

File: tests/licensedcode/data/licenses/crapl-0.1.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Lambda calculus typechecker and interpreter
+
+The code here implements a typechecker and interpreters based off three different dynamic semantics for the lambda calculus, extended with integers and four binary integer operations.  I wrote this for a class I taught at the University of Maryland, and figured I'd stick it here for posterity's sake.
+
+## License
+
+This code is released under the [CRAPL][crapl]. See [CRAPL-LICENSE.txt](CRAPL-LICENSE.txt) for more details.
```
Comment (pombredanne) [inline tests/licensedcode/data/licenses/crapl-0.1.txt]: This 
```
License
This code is released under the [CRAPL][crapl]. See [CRAPL-LICENSE.txt](CRAPL-LICENSE.txt) for more details.
```

and  that
```
This code is released under the [CRAPL][crapl]. See [CRAPL-LICENSE.txt](CRAPL-LICENSE.txt) for more details.
```

would be two good rules to add for detection.

Comment (yasharmaster) [inline tests/licensedcode/data/licenses/crapl-0.1.txt]: @pombredanne I have added these two rules.

**Comment by pombredanne** [comment]
At this stage the only thing that stills needs review and decision is the license category by @DennisClark

**Comment by pombredanne** [comment]
In particular: https://ci.appveyor.com/project/nexB/scancode-toolkit/build/640/job/sxhn62vi9gaa3g8b#L125
To run locally you can use:
`py.test -vvs -k test_detection_crapl_0_1_txt`

and this may mean that fixing would required adding more rules, e.g. possibly larger rules.

**Comment by yasharmaster** [comment]
@pombredanne I have added one more rule and this test is successful on my local machine.

**Comment by pombredanne** [comment]
@yasharmaster great. Yet I will wait  for the Ci tests to pass: eg I cannot rely only on "it works on my machine" ... See https://blog.codinghorror.com/the-works-on-my-machine-certification-program/ for some humour on the topic.

**Comment by pombredanne** [comment]
@yasharmaster Could you make the license category `Proprietary` for now? this is really what this is about from my POV.

**Comment by yasharmaster** [comment]
@pombredanne I have changed the license category to Proprietary.

**Comment by pombredanne** [comment]
Perfect. Merging. thanks!

**Comment by pombredanne** [comment]
Actually almost perfect. Please make sure you add a signed-off-by on any future commit!

**Pull request merged**